gobblin-metrics/src/main/java/gobblin/metrics/event/TimingEvent.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.event;
+
+/**
+ * Event to time actions in the program. Automatically reports start time, end time, and duration from the time
+ * the {@link gobblin.metrics.event.TimingEvent} was created to the time {@link #stop} is called.
+ */
+public class TimingEvent {
+
+  public static final String START_TIME = "startTime";
+  public static final String END_TIME = "endTime";
+  public static final String DURATION = "durationMillis";
+  public static final String TIMING_EVENT = "timingEvent";
+
+  private final String name;
+  private final Long startTime;
+  private final EventSubmitter submitter;
+  private boolean stopped;
+
+  public TimingEvent(EventSubmitter submitter, String name) {
+    this.stopped = false;
+    this.name = name;
+    this.submitter = submitter;
+    this.startTime = System.currentTimeMillis();
+  }
+
+  /**
+   * Stop the timer and submit the event.
+   */
+  public void stop() {
+    if(this.stopped) {
+      throw new RuntimeException("Timer has already been stopped.");
```

Comment:
I think it just needs to return here, and you can make clear in the Javadoc that it will be a no-op if it has already been stopped.